backend/test_observer/data_access/models_enums.py

```diff
@@ -60,3 +60,25 @@ class TestResultStatus(str, Enum):
     PASSED = "PASSED"
     FAILED = "FAILED"
     SKIPPED = "SKIPPED"
+
+
+class TestEventEnum(str, Enum):
+    STARTED_SETUP = "started_setup"
+    STARTED_PROVISION = "started_provision"
+    STARTED_FIRMWARE_UPDATE = "started_firmware_update"
+    STARTED_TEST = "started_test"
+    STARTED_ALLOCATE = "started_allocate"
+    STARTED_RESERVE = "started_reserve"
+    STARTED_CLEANUP = "started_cleanup"
+    ENDED_SETUP = "ended_setup"
+    ENDED_PROVISION = "ended_provision"
+    ENDED_FIRMWARE_UPDATE = "ended_firmware_update"
+    ENDED_TEST = "ended_test"
+    ENDED_ALLOCATE = "ended_allocate"
+    ENDED_RESERVE = "ended_reserve"
+    ENDED_CLEANUP = "ended_cleanup"
+    CANCELLED = "cancelled"
+    GLOBAL_TIMEOUT = "global_timeout"
+    OUTPUT_TIMEOUT = "output_timeout"
+    RECOVERY_FAILED = "recovery_failed"
+    FAILED = "failed"
```

Comment:
TO should ideally not know the events by name as knowing them would couple it to TF. So it would be better if there is no enum